bigframes/core/sql.py

```diff
@@ -169,3 +169,47 @@ def ordering_clause(
         part = f"`{ordering_expr.id}` {asc_desc} {null_clause}"
         parts.append(part)
     return f"ORDER BY {' ,'.join(parts)}"
+
+
+def create_vector_search_sql(
+    sql_string: str,
+    options: Mapping[str, Union[str, int, float, Iterable[str]]] = {},
+) -> str:
+    """Encode the VECTOR SEARCH statement for BigQuery Vector Search."""
+
+    base_table = options["base_table"]
+    column_to_search = options["column_to_search"]
+    distance_type = options["distance_type"]
+    top_k = options["top_k"]
+    query_column_to_search = options.get("query_column_to_search", None)
+
+    if query_column_to_search is not None:
+        query_str = f"""
+    SELECT
+        query.*,
+        base.*,
+        distance,
+    FROM VECTOR_SEARCH(
+        TABLE `{base_table}`,
+        "{column_to_search}",
```

Comment:
I think it's possible for column names to have quote characters.

Let's use `simple_literal()` to make sure we do the right escaping for all of the options values https://github.com/googleapis/python-bigquery-dataframes/blob/ea5d367d5ecc6826d30082e75c957af8362c9e61/bigframes/core/sql.py#L36

Reply:
Even if column names don't allow such characters, we don't want `vector_search()` to accidentally expose our users to SQL injection attacks.